src/main/java/fi/aalto/cs/apluscourses/presentation/base/BaseListViewModel.java

```diff
@@ -25,13 +29,18 @@
  *            {@link ListElementViewModel}
  */
 public class BaseListViewModel<E extends ListElementViewModel<?>> extends AbstractListModel<E>
```

Comment:
It crossed my mind (yes, in the middle of a night) that this class, perhaps, should no longer extend Swing's `ListModel`. That made me also think there might be a bug in this code related to the functionality of ListModel used in this class: I'm afraid that functionality has no effect any more when objects of this class ain't given to ListView's setModel method.

A solution could be making a new implementation of ListModel, working as an adapter between this class and ListView.

I just wrote these thoughts quickly up so I don't forget the issue. This needs further investigation.